include/swift/AST/Attr.h

```diff
@@ -64,8 +64,22 @@ class TypeAttributes {
   /// AtLoc - This is the location of the first '@' in the attribute specifier.
   /// If this is an empty attribute specifier, then this will be an invalid loc.
   SourceLoc AtLoc;
-  Optional<StringRef> convention = None;
-  Optional<StringRef> conventionWitnessMethodProtocol = None;
+
+  struct Convention {
+    StringRef Name = {};
+    StringRef WitnessMethodProtocol = {};
+    StringRef ClangType = {};
+
+    /// Convenience factory function to create a Swift convention.
+    ///
+    /// Don't use this function if you are creating a C convention as you
+    /// probably need a ClangType field as well.
+    static Convention makeSwiftConvention(StringRef name) {
+      return {name, "", ""};
+    }
+  };
+
+  Optional<Convention> ConventionOf;
```

Comment:
We don't use this as a naming convention; maybe `ConventionData` or `ConventionArguments`?

Reply:
There was another field named similarly

```
  Optional<OpaqueReturnTypeRef> OpaqueReturnTypeOf;
```

right below it. I'll rename it to `ConventionArguments`.